release/notes.py

```diff
@@ -0,0 +1,172 @@
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Usage: python notes.py <version> > RELEASE_NOTES.html
+
+Generates release notes for a release in HTML format containing
+introductory information about the release with links to the
+Kafka docs and the list of issues resolved in the release.
+
+The script will fail if there are any unresolved issues still
+marked with the target release. This script should be run after either
+resolving all issues or moving outstanding issues to a later release.
+"""
+
+from jira import JIRA
+import itertools, sys
+
+
+JIRA_BASE_URL = 'https://issues.apache.org/jira'
+MAX_RESULTS = 100 # This is constrained for cloud instances so we need to fix this value
+
+
+def query(query, **kwargs):
+    """
+    Fetch all issues matching the JQL query from JIRA and expand paginated results.
+    Any additional keyword arguments are forwarded to jira.search_issues.
+    """
+    results = []
+    startAt = 0
+    new_results = None
+    jira = JIRA(JIRA_BASE_URL)
+    while new_results is None or len(new_results) == MAX_RESULTS:
+        new_results = jira.search_issues(query, startAt=startAt, maxResults=MAX_RESULTS, **kwargs)
+        results += new_results
+        startAt += len(new_results)
+    return results
+
+
+def filter_unresolved(issues):
+    """
+    Some resolutions, including a lack of resolution, indicate that
+    the bug hasn't actually been addressed and we shouldn't even
+    be able to create a release until they are fixed
+    """
+    UNRESOLVED_RESOLUTIONS = [None,
+                              "Unresolved",
+                              "Duplicate",
+                              "Invalid",
+                              "Not A Problem",
+                              "Not A Bug",
+                              "Won't Fix",
+                              "Incomplete",
+                              "Cannot Reproduce",
+                              "Later",
+                              "Works for Me",
+                              "Workaround",
+                              "Information Provided"
+                              ]
+    return [issue for issue in issues if issue.fields.resolution in UNRESOLVED_RESOLUTIONS or issue.fields.resolution.name in UNRESOLVED_RESOLUTIONS]
+
+
+def issue_link(issue):
+    """
+    Generates a link to the specified JIRA issue.
+    """
+    return "%s/browse/%s" % (JIRA_BASE_URL, issue.key)
```

Comment:
I see there are a few places where you still use the `%` syntax for string interpolation. Should we use f-strings everywhere?